client_metrics.go

```diff
@@ -31,6 +34,30 @@ type ClientMetrics struct {
 	clientStreamSendHistogram        *prom.HistogramVec
 }
 
+func (m *ClientMetrics) safeClone() *ClientMetrics {
```

Comment:
what benefit does a clone give us, considering that we are copying pointer references ?
If the objective is syncing over access to the referenced values, doesn't the creation of yet another lock object defies it ? 

Acquiring the lock on the cloned object doesn't guarantee exclusive access to the pointers referenced by the original and cloned objects

Reply:
The only thing we have to protect is the values inside ClientMetrics. The prometheus types referenced by them, are all thread-safe.

We write-lock ClientMetrics where writes will happen.
In all other places, we read-lock and clone the fields, so we can pass it deeper in the code without having to worry about locking.